src/utils.rs

```diff
@@ -47,6 +47,11 @@ impl CurvePoint {
     /// key
     pub const IDENTITY: Self = CurvePoint(k256::ProjectivePoint::IDENTITY);
 
+    /// Add a curvepoint to `self`.
+    pub(crate) fn add(&self, other: &CurvePoint) -> CurvePoint {
+        CurvePoint(self.0 + other.0)
+    }
+
```

Comment:
todo: We actually implement `std::ops::Add` a bit below this; you should be able to use the `+` operator directly to add together `CurvePoint`s!